nixos/modules/services/torrent/qbittorrent-nox.nix

```diff
@@ -0,0 +1,102 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.qbittorrent-nox;
+in
+{
+  meta = {
+    doc = ./qbittorrent-nox.md;
+    maintainers = with lib.maintainers; [ redhawk ];
+  };
+
+  options.services.qbittorrent-nox = {
+    enable = lib.mkEnableOption "qBittorrent-nox headless";
+
+    dataDir = lib.mkOption {
+      type = lib.types.path;
+      default = "/var/lib/qbittorrent-nox";
+      description = "The directory where qBittorrent-nox stores its data files.";
+    };
+
+    openFirewall = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      description = "Open services.qbittorrent-nox.port to the outside network.";
+    };
+
+    user = lib.mkOption {
+      type = lib.types.str;
+      default = "qbittorrent-nox";
+      description = "User account under which qBittorrent-nox runs.";
+    };
+
+    group = lib.mkOption {
+      type = lib.types.str;
+      default = "qbittorrent-nox";
+      description = "Group under which qBittorrent-nox runs.";
+    };
+
+    port = lib.mkOption {
+      type = lib.types.port;
+      default = 8080;
+      description = "qBittorrent-nox web UI port.";
+    };
+
+    package = lib.mkPackageOption pkgs "qbittorrent-nox" { };
+  };
+
+  config = lib.mkIf cfg.enable {
+    networking.firewall = lib.mkIf cfg.openFirewall { allowedTCPPorts = [ cfg.port ]; };
+
+    systemd = {
+      services.qbittorrent-nox = {
+        description = "qBittorrent-nox service";
+        documentation = [ "man:qbittorrent-nox(1)" ];
+        after = [ "network.target" ];
+        wantedBy = [ "multi-user.target" ];
+
+        serviceConfig = {
+          User = cfg.user;
+          Group = cfg.group;
+
+          ExecStart = "${cfg.package}/bin/qbittorrent-nox --webui-port=${toString cfg.port}";
+
+          IOSchedulingClass = "idle";
+          IOSchedulingPriority = "7";
+
```

Comment:
Just a question, what effect does this have?

Reply:
Like all scheduling, either massive or no changes. But it seems like a nice to have.